v2/selective.go

```diff
@@ -38,17 +40,36 @@ func MaxTraversalLinks(MaxTraversalLinks uint64) Option {
 	}
 }
 
+// WithV1Size sets the expected v1 size of the car being written if it is known in advance.
+func WithV1Size(size uint64) Option {
```

Comment:
```suggestion
func WithDataPayloadSize(size uint64) Option {
```
The term `DataPayload` is used to describe the inner CARv1 in [CARv2 spec](https://ipld.io/specs/transport/car/carv2/#carv1-data-payload). 
It's also a term used across other v2 APIs.